okta/requestExecutor.go

```diff
@@ -23,18 +23,23 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"reflect"
+
+	"github.com/okta/okta-sdk-golang/okta/cache"
 )
 
 type RequestExecutor struct {
 	httpClient *http.Client
 	config     *Config
 	BaseUrl    *url.URL
+	cache      cache.Cache
 }
 
-func NewRequestExecutor(httpClient *http.Client, config *Config) *RequestExecutor {
+func NewRequestExecutor(httpClient *http.Client, cache cache.Cache, config *Config) *RequestExecutor {
 	re := RequestExecutor{}
 	re.httpClient = httpClient
 	re.config = config
+	re.cache = cache
 
 	if httpClient == nil {
 		re.httpClient = &http.Client{}
```

Comment:
This feels like a good opportunity to configure out default http client by taking `okta.client.requestTimeout` and apply it as `httpClient.Timeout`.  It also looks like we can take `okta.client.connectionTimeout` and use that as the idle timeout via a `Transport`:

```
tr := &http.Transport{
	IdleConnTimeout:    30 * time.Second
}
client := &http.Client{Transport: tr}
```

Reply:
configured!